### PR TITLE
Add GET:sites for AdSense module.

### PR DESCRIPTION
**Pull request by asvinb**
## Summary

Addresses issue:

- #4754 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Comment by techanvil** [comment]
Hi @asvinb, the code looks good and works well in my testing. However, the IB specifies to add tests for the new endpoint, so I'd expect to see some test coverage using `WP_REST_Request` to retrieve the endpoint, as seen in various other endpoint tests in the PHP integration suite.

I realise the other endpoints in this particular file don't have tests but maybe we can start here and backfill when we can :)

**Review by techanvil — CHANGES_REQUESTED** [review]
Hi @asvinb, the code looks good and works well in my testing. However, the IB specifies to add tests for the new endpoint, so I'd expect to see some test coverage using `WP_REST_Request` to retrieve the endpoint, as seen in various other endpoint tests in the PHP integration suite.

I realise the other endpoints in this particular file don't have tests but maybe we can start here and backfill when we can :)

**Comment by asvinb** [comment]
Thanks @techanvil . @aaemnnosttv @eugene-manuilov Is there a reason why we don't have any PHP tests for module endpoints? I see we are only checking for the datapoints, but that's it. I wondering whether it's an oversight or there was a specific reason since none endpoints are tested.

**Comment by eugene-manuilov** [comment]
@techanvil @asvinb I think the only test that we can add for the new endpoint is the one that checks that the endpoint returns the appropriate error if the request doesn't contain the `accountID` value. No need to check anything else there for now. We have some ideas on how to refactor endpoints and make them easier to test, but it's still yet to be implemented.

> Is there a reason why we don't have any PHP tests for module endpoints? I see we are only checking for the datapoints, but that's it. I wondering whether it's an oversight or there was a specific reason since none endpoints are tested.

I can't say exactly because the majority of datapoints had been already implemented before I joined the project. My hunch is that we didn't add it because of the complexity of those tests because we would need to mock a lot of functionality from the Google SDK that we use in our endpoints.

File: tests/phpunit/integration/Modules/AdSenseTest.php [hunk 1]
```diff
@@ -529,4 +535,47 @@ public function test_parse_earnings_orderby() {
 		$this->assertEquals( '+sessions', $result[1] );
 	}
 
+	public function test_get_sites_no_account_ID_REST_endpoint() {
```
[the remark this answers is not in the document]
Comment (asvinb) [reply tests/phpunit/integration/Modules/AdSenseTest.php]: Good catch @techanvil ! Updated!

File: tests/phpunit/integration/Modules/AdSenseTest.php [hunk 2]
```diff
@@ -529,4 +535,47 @@ public function test_parse_earnings_orderby() {
...
+		$options        = new Options( $context );
+		$user_options   = new User_Options( $context, $user->ID );
+		$authentication = new Authentication( $context, $options, $user_options );
+		$adsense        = new AdSense( $context, $options, $user_options, $authentication );
+		$adsense->register();
+
+		$authentication->get_oauth_client()->set_granted_scopes(
+			$adsense->get_scopes()
+		);
+		$request  = new WP_REST_Request( 'GET', '/' . REST_Routes::REST_ROOT . '/modules/adsense/data/sites?accountID=foo' );
```
[the remark this answers is not in the document]
Comment (asvinb) [reply tests/phpunit/integration/Modules/AdSenseTest.php]: I've removed the test, however I'am thinking maybe we should test for a return code other than `400` but like you said, if we are putting extra load on the servers, then we can wait.

[the remark this answers is not in the document]
Comment (techanvil) [reply tests/phpunit/integration/Modules/AdSenseTest.php]: Nice one @asvinb :ok_hand:

**Review by techanvil — APPROVED** [review]
LGTM, nice one @asvinb

**Review by eugene-manuilov — APPROVED** [review]
LGTM too

**Pull request merged**